mesonbuild/compilers/compilers.py

```diff
@@ -510,7 +510,7 @@ def __repr__(self) -> str:
     def can_compile(self, src: 'mesonlib.FileOrString') -> bool:
         if isinstance(src, mesonlib.File):
             src = src.fname
-        suffix = os.path.splitext(src)[1].lower()
+        suffix = os.path.splitext(src)[1]
```

Comment:
I have a nagging feeling that this will break things for someone. It also requires adding things in many different places. How about just special casing the upper case C? Something like:

```python
suffix = os.path.splitext(src)[1]
if suffix != 'C':
    suffix = suffix.lower()

Reply:
Ok, I did that.